### PR TITLE
move to rust 2021

### LINKED ISSUE #4277: Support Rust 2021 edition
Building on https://github.com/paritytech/substrate/pull/10170 , moving to 2021 throughout this project will be :kissing_cat: :ok_hand:

### PR DESCRIPTION
**Pull request by NukeManDan**
close #4277 
Confirmed to build locally on Ubuntu 20.04.3 LTS:
```
rustc 1.56.1 (59eed8a2a 2021-11-01)
rustc 1.58.0-nightly (e90c5fbbc 2021-11-12)
```

### CODE REVIEW

**Review by KiChjang — APPROVED** [review]

**Comment by NukeManDan** [comment]
`rust-version = "1.56.1"` introduced

For reference, I tried to get this to `cargo build` with 1.56.0 and it failed with
```
error: package `...` cannot be built because it requires rustc 1.56.1 or newer, while the currently active rustc version is 1.56.0
```
With `--ignore-rust-version` it _does_ build with 1.56.0, as expected.

**Review by KiChjang — APPROVED** [review]
it still gud

**Review by Jaysonmaldx — APPROVED** [review]
Thank u

File: bridges/bin/millau/node/Cargo.toml [hunk 1]
```diff
@@ -3,7 +3,8 @@ name = "millau-bridge-node"
 description = "Substrate node compatible with Millau runtime"
 version = "0.1.0"
 authors = ["Parity Technologies <admin@parity.io>"]
-edition = "2018"
+edition = "2021"
+rust-version = "1.56.1"
```
Comment (bkchr) [inline bridges/bin/millau/node/Cargo.toml:7]: Please don't touch any bridges code here. That needs to be done upstream.

**Review by bkchr — CHANGES_REQUESTED** [review]

**Comment by drahnr** [comment]
Closing, migration already happened in `645ae551fcce` .

**Pull request closed without merging**